test/test_support.ts

```diff
@@ -1,42 +1,92 @@
+  let p = ts.getDefaultLibFilePath(OPTIONS);
+  return {cachedLibPath: p, cachedLib: host.getSourceFile(fn, ts.ScriptTarget.ES6)};
 })();
 
-function transformSource(src: string): string {
+function annotateSource(src: string): string {
   var host = ts.createCompilerHost(OPTIONS);
   var original = host.getSourceFile.bind(host);
   host.getSourceFile = function(fileName: string, languageVersion: ts.ScriptTarget,
                                 onError?: (msg: string) => void): ts.SourceFile {
-    if (fileName === cachedLibName) return cachedLib;
+    if (fileName === cachedLibPath) return cachedLib;
     if (fileName === 'main.ts') {
       return ts.createSourceFile(fileName, src, ts.ScriptTarget.Latest, true);
     }
     return original(fileName, languageVersion, onError);
   };
 
-  var program = ts.createProgram(['main.ts'], {}, host);
+  var program = ts.createProgram(['main.ts'], OPTIONS, host);
   if (program.getSyntacticDiagnostics().length) {
     throw new Error(formatDiagnostics(ts.getPreEmitDiagnostics(program)));
   }
 
-  var res = transformProgram(program);
+  var res = annotateProgram(program);
   expect(Object.keys(res)).to.deep.equal(['main.ts']);
   return res['main.ts'];
 }
 
+function transformSource(src: string): string {
+  var host = ts.createCompilerHost(OPTIONS);
+  var original = host.getSourceFile.bind(host);
+  var mainSrc = ts.createSourceFile('main.ts', src, ts.ScriptTarget.Latest, true);
+  host.getSourceFile = function(fileName: string, languageVersion: ts.ScriptTarget,
+                                onError?: (msg: string) => void): ts.SourceFile {
+    if (fileName === cachedLibPath) return cachedLib;
+    if (fileName === 'main.ts') {
+      return mainSrc;
+    }
+    return original(fileName, languageVersion, onError);
+  };
+
+  var program = ts.createProgram(['main.ts'], OPTIONS, host);
+  if (program.getSyntacticDiagnostics().length) {
+    throw new Error(formatDiagnostics(ts.getPreEmitDiagnostics(program)));
+  }
+
+  var transformed: StringMap = {};
+  var emitRes =
+      program.emit(mainSrc, (fileName: string, data: string) => { transformed[fileName] = data; });
+  if (emitRes.diagnostics.length) {
+    throw new Error(formatDiagnostics(emitRes.diagnostics));
+  }
+  expect(Object.keys(transformed)).to.deep.equal(['main.js']);
+  return transformed['main.js'];
+}
+
 export function expectSource(src: string) {
-  return expect(transformSource(src));
+  var annotated = annotateSource(src);
+  // console.log('Annotated', annotated);
+  var transformed = transformSource(annotated);
+  return expect(transformed);
+}
+
+export interface GoldenFileTest { name: string, tsPath: string, jsPath: string, }
+
+export function goldenTests(): GoldenFileTest[] {
+  var tsExtRe = /\.ts$/;
+  var testFolder = path.join(__dirname, '..', '..', 'test_files');
+  var files = fs.readdirSync(testFolder).filter((fn) => !!fn.match(tsExtRe));
+  return files.map((fn) => {
```

Comment:
You'd think - but not if what you're returning is an object literal:

This code gets parsed as a block with a goto label `foo:` and a sole property access `bar`:

```js
var fn = (x) => { foo: bar };
```